x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlExecutionInfo.java

```diff
@@ -70,6 +70,7 @@ public class EsqlExecutionInfo implements ChunkedToXContentObject, Writeable {
     private final transient Predicate<String> skipUnavailablePredicate;
     private final transient Long relativeStartNanos;  // start time for an ESQL query for calculating took times
     private transient TimeValue planningTookTime;  // time elapsed since start of query to calling ComputeService.execute
+    private transient boolean isPartial; // Does this request have partial results?
```

Comment:
I think this is the right thing to do, but we'll want to sync with @luigidellaquila on how "general purpose" partial results will work and if we'll be setting some "is_partial" flag on the response.

Reply:
Yes I am not sure also this partial and `Cluster.Status.PARTIAL` are related... Need to figure this out yet.